ocaml/xapi-cli-server/cli_operations.ml

```diff
@@ -4058,12 +4063,12 @@ let vm_install printer rpc session_id params =
           List.fold_left filter_records_on_fields all_recs
             (("name-label", name) :: filter_params)
         in
-        match List.length templates with
-        | 0 ->
+        match templates with
+        | [] ->
             failwith "No templates matched"
-        | 1 ->
-            (List.hd templates).getref ()
-        | _ ->
+        | [x] ->
+            x.getref ()
+        | _ :: _ ->
```

Comment:
Technically, `[x]` is `x :: []`, which matches the pattern `_ :: _`, which makes it a bit awkward. If you want to be specific, `_ :: _ :: _` should be used, or `_` is fine as well.

Reply:
I was relying on the order here. Will change to `_ :: _ :: _` to future-proof it.